src/Amalgam/entity/EntityQueryBuilder.h

```diff
@@ -43,17 +43,133 @@ namespace EntityQueryBuilder
 			|| type == ENT_COMPUTE_ENTITY_KL_DIVERGENCES);
 	}
 
-	//TODO 17631: expand this method
+	//populates deviation data for a given nominal value
+	//assumes that value_deviation_assoc is a valid pointer to an assoc
+	template<typename NominalDeviationValuesType>
+	inline void PopulateFeatureDeviationNominalValueAssocData(
+		NominalDeviationValuesType &ndd, EvaluableNode *value_deviation_assoc)
+	{
+		auto &mcn = value_deviation_assoc->GetMappedChildNodesReference();
+		ndd.reserve(mcn.size());
+		for(auto &cn : mcn)
+		{
+			if constexpr(std::is_same<typename NominalDeviationValuesType::key_type, double>::value)
+			{
+				double value = std::numeric_limits<double>::quiet_NaN();
+				if(cn.first != string_intern_pool.EMPTY_STRING_ID)
+				{
+					auto [number_value, success] = Platform_StringToNumber(string_intern_pool.GetStringFromID(cn.first));
+					if(success)
+						value = number_value;
+				}
+
+				ndd.emplace(value, EvaluableNode::ToNumber(cn.second));
+			}
+			else
+			{
+				ndd.emplace(cn.first, EvaluableNode::ToNumber(cn.second));
+			}
+		}
+	}
+
+	//populates deviation data for a given nominal value
+	template<typename NominalDeviationValuesType>
+	inline void PopulateFeatureDeviationNominalValueData(
+		NominalDeviationValuesType &ndd, EvaluableNode *value_deviation_node)
+	{
+		if(EvaluableNode::IsEmptyNode(value_deviation_node))
+			return;
+
+		auto vdn_type = value_deviation_node->GetType();
+
+		//if it's an assoc, just populate, otherwise parse list with assoc in it
+		if(vdn_type == ENT_ASSOC)
+		{
+			PopulateFeatureDeviationNominalValueAssocData<NominalDeviationValuesType>(ndd, value_deviation_node);
+		}
+		else if(vdn_type == ENT_LIST)
```

Comment:
please comment this better re what this is expected to look like on the input / default. the code is readable, but the comment would be easier to grok and associate with the language.js file